src/main/java/ch/heigvd/pro/b04/polls/PollController.java

```diff
@@ -85,6 +86,38 @@ public ServerPoll insert(
         .orElseThrow(WrongCredentialsException::new);
   }
 
+  /**
+   * Updates an existing poll with some new information.
+   *
+   * @param token       The authentication token for the user.
+   * @param idModerator The identifier of the moderator who is updating the poll.
+   * @param idPoll      The identifier of the poll to update.
+   * @param clientPoll  The data with which the poll should be updated.
+   * @return The updated poll.
+   * @throws WrongCredentialsException If the moderator is invalid, or the credentials incorrect.
+   * @throws ResourceNotFoundException If the poll that we're trying to update does not exist.
+   */
+  @PutMapping("/mod/{idModerator}/poll/{idPoll}")
+  public ServerPoll update(
+      @RequestParam(name = "token") String token,
+      @PathVariable(name = "idModerator") Integer idModerator,
+      @PathVariable(name = "idPoll") Integer idPoll,
+      @RequestBody ClientPoll clientPoll
+  ) throws WrongCredentialsException, ResourceNotFoundException {
+
+    ServerPollIdentifier identifier = moderators.findBySecret(token)
+        .filter(moderator -> moderator.getIdModerator() == idModerator)
+        .map(moderator -> ServerPollIdentifier.builder()
+            .idxModerator(moderator)
+            .idPoll(idPoll)
+            .build())
+        .orElseThrow(WrongCredentialsException::new);
+
+    polls.update(identifier, clientPoll.getTitle());
```

Comment:
Do I understand correctly that this method will not throw an exception if the poll doesn't exist ?